src/accelerate/commands/estimate.py

```diff
@@ -0,0 +1,270 @@
+    """
+    Creates an empty model from its parent library on the `Hub` to calculate the overall memory consumption.
+
+    Args:
+        model_name (`str`):
+            The model name on the Hub
+        library_name (`str`):
+            The library the model has an integration with, such as `transformers`. Will be used if `model_name` has no
+            metadata on the Hub to determine the library.
+        trust_remote_code (`bool`, `optional`, defaults to `False`):
+            Whether or not to allow for custom models defined on the Hub in their own modeling files. This option
+            should only be set to `True` for repositories you trust and in which you have read the code, as it will
+            execute code present on the Hub on your local machine.
+
+    """
+    model_info = verify_on_hub(model_name)
+    # Simplified errors
+    if model_info == "gated":
+        raise GatedRepoError(f"Repo for model `{model_name}` is gated. You must be authenticated to access it.")
+    elif model_info == "repo":
+        raise RepositoryNotFoundError(
+            f"Repo for model `{model_name}` does not exist on the Hub. If you are trying to access a private repo, make sure you are authenticated."
+        )
+    if library_name is None:
+        library_name = getattr(model_info, "library_name", False)
+        if not library_name:
+            raise ValueError(
+                f"Model `{model_name}` does not have any library metadata on the Hub, please manually pass in a `--library_name` to use (such as `transformers`)"
+            )
+    if library_name == "transformers":
+        if not is_transformers_available():
+            raise ImportError(
+                f"To check `{model_name}`, `transformers` must be installed. Please install it via `pip install transformers`"
+            )
+        print(f"Loading pretrained config for `{model_name}` from `transformers`...")
+        # As we just load in the `config` file and not the weights, we use `True`
+        auto_map = model_info.config.get("auto_map", False)
+        if auto_map and not trust_remote_code:
+            raise ValueError(
+                f"Loading {model_name} requires to execute some code in that repo, you can inspect the content of "
+                f"the repository at https://hf.co/{model_name}. You can bypass this by passing "
+                "`--trust_remote_code`"
+            )
+
+        try:
+            config = AutoConfig.from_pretrained(model_name, trust_remote_code=trust_remote_code)
+        except ValueError as e:
+            if "trust_remote_code" in e.args[0]:
+                e.args = (
+                    f"Loading {model_name} requires to execute some code in that repo, you can inspect the content of "
+                    f"the repository at https://hf.co/{model_name}. You can bypass this by passing "
+                    "`--trust_remote_code`",
+                )
+            raise e
+        with init_empty_weights():
```

Comment:
I'd let the error be raised by Transformers and just do the config instantiation here.